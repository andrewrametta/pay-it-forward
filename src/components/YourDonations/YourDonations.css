div.yourdonations-container {
  height: 100%;
  padding: 30px 0px;
}

li.your-items-div-container {
  margin: 50px 10px;
  background: white;
  padding: 10px;
  box-shadow: 2px 2px 6px 0 rgba(41, 2, 80, 0.1);
  border-radius: 10px;
}

div.yourdonations-header {
  text-align: center;
}

div.your-item-img {
  text-align: center;
}

div.your-items-container {
  height: 100%;
  padding-bottom: 50px;
}

img.img-yourdonation {
  width: 100%;
  height: 100%;
  padding: 5px;
}

@media all and (min-width: 400px) {
  .yourdonations-container {
    margin-left: 10%;
    margin-right: 10%;
  }

  ul.yourdonations-items {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-around;
    padding: 10px;
  }

  li.your-items-div-container {
    width: 250px;
    margin-bottom: 20px;
    padding: 10px;
    flex: 1;
  }

  img.img-yourdonation {
    width: 200px;
    height: 200px;
    padding: 5px;
  }
}
