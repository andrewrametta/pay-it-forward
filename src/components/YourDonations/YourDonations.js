import React, { useContext, useEffect, useState } from "react";
import Donation from "../Donation/Donation";
import AppContext from "../../AppContext";
import AuthAPIService from "../../services/auth-api-service";

export default function YourDonations(props) {
  const [error, setError] = useState("");
  const { items, setItems, userId } = useContext(AppContext);

  useEffect(() => {
    AuthAPIService.getItems()
      .then((donations) => {
        setItems(donations);
      })
      .catch((res) => {
        setError(error);
      });
  }, []);

  return (
    <div className="dashboard-container">
      <h1>Donations Available</h1>
      <div className="dashboard-item-container">
        <ul className="ul-items">
          {items
            .filter((item) => item.user_id === userId)
            .map((filteredItem, indx) => (
              <Donation
                key={indx}
                id={filteredItem.id}
                src={filteredItem.item_url}
                title={filteredItem.title}
                description={filteredItem.description}
              />
            ))}
        </ul>
      </div>
    </div>
  );
}
