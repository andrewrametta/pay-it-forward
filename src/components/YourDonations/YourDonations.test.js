import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import YourDonations from "./YourDonations";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AppContext from "../../AppContext";

configure({ adapter: new Adapter() });

describe(`YourDonations component`, () => {
  it.only("renders the complete page", () => {
    const wrapper = shallow(
      <AppContext.Provider>
        <YourDonations />
      </AppContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
