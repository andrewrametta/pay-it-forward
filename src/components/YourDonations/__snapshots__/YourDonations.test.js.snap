// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourDonations component renders the complete page 1`] = `<YourDonations />`;
